### PR TITLE
fix: remove exceptions from balance logic library

### PR DESCRIPTION
**Pull request by lekhovitsky**

### CODE REVIEW

File: contracts/credit/CreditFacadeV3.sol [hunk 1]
```diff
@@ -824,7 +824,8 @@ contract CreditFacadeV3 is ICreditFacadeV3, ACLNonReentrantTrait {
         // If expectedBalances was set by calling revertIfGetLessThan,
         // checks that actual token balances are not less than expected balances
         if (expectedBalances.length != 0) {
-            BalancesLogic.compareBalances(creditAccount, expectedBalances); // U:[FA-23]
+            bool violation = BalancesLogic.compareBalances(creditAccount, expectedBalances);
```
Comment (0xmikko) [inline contracts/credit/CreditFacadeV3.sol]: as for me, `bool success` is more common practise

File: contracts/libraries/BalancesLogic.sol [hunk 2]
```diff
@@ -1,10 +1,9 @@
 // SPDX-License-Identifier: MIT
 // Gearbox Protocol. Generalized leverage for DeFi protocols
-// (c) Gearbox Holdings, 2022
+// (c) Gearbox Holdings, 2023
```
Comment (0xmikko) [inline contracts/libraries/BalancesLogic.sol:3]: Maybe, it's better to update the whole codebase with one action?

File: contracts/credit/CreditFacadeV3.sol (hunk 1 above)
Comment (lekhovitsky) [reply contracts/credit/CreditFacadeV3.sol]: makes sense, but "if (success) revert" seems confusing :)

File: contracts/libraries/BalancesLogic.sol (hunk 2 above)
Comment (lekhovitsky) [reply contracts/libraries/BalancesLogic.sol:3]: let's call it a built-in "format-on-save" hook in my brain

File: contracts/credit/CreditFacadeV3.sol (hunk 1 above)
Comment (0xmikko) [reply contracts/credit/CreditFacadeV3.sol]: what if....? I suggest...
```
 bool success = BalancesLogic.compareBalances(creditAccount, expectedBalances);
 if (!success) revert(...)
 ```

**Review by 0xmikko — APPROVED** [review]

File: contracts/libraries/BalancesLogic.sol [hunk 3]
```diff
@@ -30,30 +28,30 @@ library BalancesLogic {
...
         }
     }
 
     /// @dev Compares current balances to previously saved expected balances.
-    /// Reverts if at least one balance is lower than expected
     /// @param creditAccount Credit Account to check
     /// @param expected Expected balances after all operations
-    function compareBalances(address creditAccount, Balance[] memory expected) internal view {
-        uint256 len = expected.length; // F:[FA-45]
+    /// @return False if at least one balance is lower than expected, true otherwise
```
Comment (0xmikko) [inline contracts/libraries/BalancesLogic.sol]: Probably, named return could help others to understand the logic

**Review by 0xmikko — APPROVED** [review]

File: contracts/libraries/BalancesLogic.sol (hunk 3 above)
Comment (lekhovitsky) [reply contracts/libraries/BalancesLogic.sol]: done

**Pull request merged**